pkgs/by-name/ap/apple-sdk/common/run-build-phase-hooks.nix

```diff
@@ -0,0 +1,10 @@
+{ }:
+
+self: super:
+{
+  buildPhase = ''
+    runHook preBuild
+    ${super.buildPhase or ""}
+    runHook postBuild
+  '';
+}
```

Comment:
I tried other patterns, but they made maintainability worse. What would be good for evaluation performance is figuring out why the SDK is reevaluated hundreds of times instead of far fewer than that.